model_analyzer/monitor/nvml.py

```diff
@@ -66,23 +67,26 @@ def _monitoring_loop(self, tags):
         self._thread_active = True
         frequency = self._frequency
 
-        record_collector = RecordCollector()
+        records = []
         while self._thread_active:
             begin = time.time()
             for tag in tags:
                 if tag == 'memory':
                     for i, handle in enumerate(self._nvml_handles):
                         memory = nvmlDeviceGetMemoryInfo(handle)
                         gpu_device = self._gpus[i]
-                        memory_record = GPUMemoryRecord(
-                            gpu_device, memory.used, memory.total, memory.free)
-                        record_collector.insert(memory_record)
+                        records.append(
+                            GPUUsedMemory(device=gpu_device,
+                                          used_mem=memory.used))
```

Comment:
How does this work? The __init__ method for GPUUsedMemory and GPUFreeMemory requires a timestamp.

Reply:
I thought I set a default value at the very least. Not sure how this happened. Must have passed the test pre-merge conflict resolution.